### PR TITLE
APPS-14 Add events_row_id to event annotations.

### PR DESCRIPTION
**Pull request by terloon**
## 📚 Purpose
*Add events_row_id as a foreign key to event_annotations table to associate with events.*

## 👌 Resolves:
- [ ] APPS-14 -implements this

## 📦 Impacts:
*mds-ingest-service*

## ✅ PR Checklist
- [ ] Event annotations are now associated with events via events_row_id.

### CODE REVIEW

File: packages/mds-ingest-service/tests/index.spec.ts [hunk 1]
```diff
@@ -550,6 +550,18 @@ describe('Ingest Service Tests', () => {
...
+    it('gets events with 2 annotations', async () => {
+      await IngestRepository.createDevices([TEST_TNC_A, TEST_TNC_B])
+      await IngestRepository.createEvents([TEST_EVENT_A1, TEST_EVENT_B1])
+      await IngestRepository.createEvents([TEST_EVENT_A2, TEST_EVENT_B2])
+      await IngestServiceClient.writeEventAnnotations([TEST_EVENT_ANNOTATION_A, TEST_EVENT_ANNOTATION_B])
+      const { events } = await IngestServiceClient.getEventsUsingOptions({
+        time_range: { start: testTimestamp, end: testTimestamp + 2000 },
+        grouping_type: 'all_events'
+      })
+      expect(events.filter(e => e.annotation).length).toEqual(2)
```
Comment (nessur) [inline packages/mds-ingest-service/tests/index.spec.ts:563]: good use of `.filter()` ! :)

File: packages/mds-ingest-service/repository/entities/event-annotation-entity.ts [hunk 2]
```diff
@@ -47,6 +48,13 @@ export class EventAnnotationEntity extends IdentityColumn(RecordedColumn(class {
 
   @Column('bigint', { transformer: BigintTransformer })
   latency_ms: Timestamp
+
+  @OneToOne(type => EventEntity, event => event.annotation, { nullable: true })
```
Comment (BigIdeas) [inline packages/mds-ingest-service/repository/entities/event-annotation-entity.ts:52]: I think the `nullable` is flipped on these new properties. An EventAnnotation will always have an Event but an Event may not always have an EventAnnotation.

File: packages/mds-ingest-service/repository/entities/event-annotation-entity.ts [hunk 3]
```diff
@@ -47,6 +48,13 @@ export class EventAnnotationEntity extends IdentityColumn(RecordedColumn(class {
...
+  @JoinColumn({ name: 'events_row_id', referencedColumnName: 'id' })
+  event?: EventEntityModel
+
+  @Column('bigint')
+  events_row_id?: number
```
Comment (avatarneil) [inline packages/mds-ingest-service/repository/entities/event-annotation-entity.ts:57]: And this should always exist, right?

Comment (drtyh2o) [reply packages/mds-ingest-service/repository/entities/event-annotation-entity.ts:57]: Also, `bigint` columns should have the `{ transformer: BigintTransformer }` option

File: packages/mds-ingest-service/repository/entities/event-annotation-entity.ts (hunk 2 above)
Comment (terloon) [reply packages/mds-ingest-service/repository/entities/event-annotation-entity.ts:52]: Looking into this.

Comment (terloon) [reply packages/mds-ingest-service/repository/entities/event-annotation-entity.ts]: Made event and events_row_id non-nullable.

File: packages/mds-ingest-service/repository/entities/event-annotation-entity.ts (hunk 3 above)
Comment (terloon) [reply packages/mds-ingest-service/repository/entities/event-annotation-entity.ts]: Added transformer.

File: packages/mds-ingest-service/tests/index.spec.ts [hunk 4]
```diff
@@ -191,6 +191,30 @@ const TEST_EVENT_ANNOTATION_B: EventAnnotationDomainCreateModel = {
...
+  device_id: DEVICE_UUID_B,
+  timestamp: testTimestamp,
+  vehicle_id: 'test-id-2',
+  vehicle_type: 'scooter' as VEHICLE_TYPE,
+  propulsion_types: ['electric'] as PROPULSION_TYPE[],
+  geography_ids: [uuid(), uuid()],
+  geography_types: [null, 'spot'],
+  latency_ms: 150
+}
+
```
Comment (BigIdeas) [inline packages/mds-ingest-service/tests/index.spec.ts:217]: You should be able to just do
```suggestion
const TEST_EVENT_ANNOTATION_C = {
  ...TEST_EVENT_ANNOTATION_A,
  events_row_id: 1
}

const TEST_EVENT_ANNOTATION_D = {
  ...TEST_EVENT_ANNOTATION_B,
  events_row_id: 2
}
```

**Review by BigIdeas — APPROVED** [review]
Nice!

**Review by avatarneil — APPROVED** [review]

**Pull request merged**